modules/viewabilityScoreGeneration.js

```diff
@@ -15,10 +16,64 @@ const GPT_SLOT_RENDER_ENDED_EVENT = 'slotRenderEnded';
 const GPT_IMPRESSION_VIEWABLE_EVENT = 'impressionViewable';
 const GPT_SLOT_VISIBILITY_CHANGED_EVENT = 'slotVisibilityChanged';
 const TOTAL_VIEW_TIME_LIMIT = 1000000000;
-const domain = window.location.hostname;
 const NATIVE_DEFAULT_SIZE = '0x0';
+const DEFAULT_SERVER_CALL_FREQUENCY = { metric: 'hours', duration: 6 };
 const ADSLOTSIZE_INDEX = 2;
 const ADUNIT_INDEX = 1;
+const ENDPOINT = 'http://localhost:3000/fake-endpoint/inventory-packaging' // TODO: update this to a real endpoint when ready
+const domain = window.location.hostname;
+let vsgObj;
+
+storage.getDataFromLocalStorage('viewability-data', val => {
+  vsgObj = JSON.parse(val);
+});
+
+export const fireToServer = auctionData => {
+  const payload = generatePayload(auctionData, vsgObj);
+
+  ajax(ENDPOINT, (res) => {
+    // do nothing for now
+  }, JSON.stringify(payload));
+
+  vsgObj = null;
+  removeFromLocalStorage('viewability-data');
+};
+
+export const generatePayload = (auctionData, lsObj) => {
+  const adSizeKeys = Object.keys(lsObj).filter(key => /([0-9]+x[0-9]+)/.test(key));
```

Comment:
I could not get what exactly we are doing here.

Reply:
right here we are just gathering all of the ad size keys currently in local storage using regex (ex. key names like "300x250", "728x90", etc.)